platform/security/idp/security-idp-client/src/test/java/org/codice/ddf/security/idp/client/LogoutRequestServiceTest.java

```diff
@@ -13,11 +13,15 @@
  */
 package org.codice.ddf.security.idp.client;
```

Comment:
This was a difficult class to work with. I spend day or so refactoring it. Decided I was spending too much time, so I reverted to the original version. 

Fixed up tests that broke because of session index needing to passed into constructors. Also added new tests.